samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java

```diff
@@ -1003,9 +1030,19 @@ public int compare(ItemGradingData o1, ItemGradingData o2) {
         		}
         	}
         }
-        if ((TypeIfc.IMAGEMAP_QUESTION.equals(itemType))&&(NeededAllOk)&&((autoScore==-123456789)||!imageMapAllOk)){
-        	autoScore=0;
-        	imageMapAllOk=false;
+        
+        if (TypeIfc.IMAGEMAP_QUESTION.equals(itemType) && neededAllOk){
+        	if(!imageMapAlreadyOk) {
+        		autoScore = 0;
+        	}else if(autoScore == -123456789){
```

Comment:
I know this was n't added by you but this number looks strange?

Reply:
it's just like a mark that we need all right answers and the actual answer is not right. I replaced it by a constant now.